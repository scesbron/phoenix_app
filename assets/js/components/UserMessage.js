import React from "react";
import PropTypes from "prop-types";

class UserMessage extends React.Component {
  render() {
    const { fromServer, username, message } = this.props;
    const className = fromServer
      ? "user-message user-message-server"
      : "user-message user-message-local";
    return (
      <div className={className}>
        <div className="user-message-title">{username}</div>
        <div className="user-message-text">{message}</div>
      </div>
    );
  }
}

UserMessage.propTypes = {
  fromServer: PropTypes.bool,
  username: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
};

export default UserMessage;
