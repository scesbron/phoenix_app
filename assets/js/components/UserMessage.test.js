import React from "react";
import UserMessage from "./UserMessage";
import renderer from "react-test-renderer";

it("renders correctly local message", () => {
  const tree = renderer
    .create(<UserMessage fromServer={false} username="Title" message="Bonjour" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly server message", () => {
  const tree = renderer
    .create(<UserMessage fromServer={true} username="Title" message="Bonjour" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
