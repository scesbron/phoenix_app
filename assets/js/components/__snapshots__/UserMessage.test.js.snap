// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly local message 1`] = `
<div
  className="user-message user-message-local"
>
  <div
    className="user-message-title"
  >
    Title
  </div>
  <div
    className="user-message-text"
  >
    Bonjour
  </div>
</div>
`;

exports[`renders correctly server message 1`] = `
<div
  className="user-message user-message-server"
>
  <div
    className="user-message-title"
  >
    Title
  </div>
  <div
    className="user-message-text"
  >
    Bonjour
  </div>
</div>
`;
